include/deal.II/matrix_free/operators.h

```diff
@@ -160,6 +160,12 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
+     * Get read access to the MatrixFree object stored with this operator.
+     */
+    const std_cxx11::shared_ptr<const MatrixFree<dim,Number> > &
```

Comment:
>  If you agree, I will change.

i agree. Another potential usage case could be to get the shared pointer from some operator, delete the operator object and re-use the MatrixFree data in another operator.